### ISSUE TITLE
Please add support for switch role style CLI profiles

### ISSUE THREAD

**Issue by eric-tucker** [aws/aws-sdk-php#1030]
Boto3 seems to be the only SDK that supports this:

```
[profile otheraccount]
output = json
region = us-east-1
role_arn = arn:aws:iam::234567890123:role/MyRole
source_profile = mymainprofile
```

```
export AWS_PROFILE=otheraccount && php my_script.php
```

Currently, it appears the SDK only supports profiles which are using credentials, and this doesn't work in an AWS setup using federated access, assume role, and so forth where IAM users are not an option.

**Comment by cjyclaire** [comment]
@eric-tucker Appreciate your feedback, shared configuration(including AssumeRole) across SDKs is already in out backlog, which will be in the implementation queue soon!

**Comment by cjyclaire** [comment]
Ping updates here, PR #1077 is opened in supporting config profile as well as assume role.

**Comment by cjyclaire** [comment]
Closing, tracking the PR in progress instead.

**Comment by eric-tucker** [comment]
@cjyclaire I don't believe this was resolved with #1137 .

I am still encountering issues with using an assume role from the credentials file that looks like this:
```
[myprofilethatassumesrole]
output = json
region = us-east-1
role_arn = arn:aws:iam::123456789012:role/MyAssumeRole
source_profile = myprofilewithcreds
```
I get this exception:
Fatal error: Uncaught Aws\Exception\CredentialsException: No credentials present in INI profile 'myprofilethatassumesrole' (/var/www/.aws/credentials) in /var/www/html/vendor/aws/aws-sdk-php/src/Credentials/CredentialProvider.php:394
```
I don't see in #1137 where a change was made to actually call this when the profile being used is an assume role type profile.  Having to change the code to work with assumed roles (by declaring a custom credential provider) isn't an option.
